Dockerfile

```
@@ -0,0 +1,26 @@
+FROM python:3.8.12-slim-buster@sha256:55ef3d2132dec7f372f4d63fbec0027e23388ab072533edbba9ca213a053c9cb as build
+RUN mkdir /app
+RUN mkdir /packages
+RUN useradd -m developer
+RUN groupadd auth0
+RUN usermod -a -G auth0 developer
+RUN chown -R developer:auth0 /app
+RUN chown -R developer:auth0 /packages
+WORKDIR /app
+USER developer
+COPY requirements.txt .
+RUN pip install --disable-pip-version-check -r requirements.txt --target /packages
+COPY authz authz/
+COPY common common/
+COPY hello_world hello_world/
+COPY messages_api messages_api/
+COPY gunicorn.conf.py manage.py ./
```

Should we use the `./` prefix in all of these to maintain the consistency and provide clarity that we're copying relative to the context and the `WORKDIR`? Also, the trailing slash is not needed 🙂 
```suggestion
COPY ./authz ./authz
COPY ./common ./common
COPY ./hello_world ./hello_world
COPY ./messages_api ./messages_api
COPY ./gunicorn.conf.py ./manage.py ./
```

---

@mrcrow85 it could be the other way around too (i.e. not prefixing with `./`). The idea is to maintain consistency 😁

---

@JoseLion According to [docker](https://docs.docker.com/engine/reference/builder/#copy) if the destination doesn't end with a trailing `/` docker will assume its a file and the contents will be copied at that location (without the directory), so basically with a trailing `/` we are telling docker we want a directory at that location with the contents of `<src>` because `COPY` does not copy the directory itself.

---

also this line `COPY gunicorn.conf.py manage.py ./` is correct because we need `./` when specifying more than one file on the command.